core/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -288,7 +282,7 @@ public final Searcher acquireSearcher(String source) throws EngineException {
             try {
                 final Searcher retVal = newSearcher(source, searcher, manager);
                 success = true;
-                return retVal;
+                return config().getWrappingService().wrap(retVal);
```

Comment:
Would EngineSearcherFactory be a better place to do the wrapping? Otherwise I'm concerned we might not call incRef() and decRef() on the same index readers.